### PR TITLE
Removed redundant code from layout_optimizer.cc

### PR DESCRIPTION
**Pull request by joyalbin**

### CODE REVIEW

**Review by andyly — COMMENTED** [review]
These calls are necessary for creating Const nodes holding the permutation used by added Transposes of nodes not in a frame.

**Comment by joyalbin** [comment]
@andyly Thank you so much for making me understand this logic more clearly.
I have added one line of comment to make the code more understandable.
Could you please review the change.

File: tensorflow/core/grappler/optimizers/layout_optimizer.cc [hunk 1]
```diff
@@ -2048,8 +2048,11 @@ class DataLayoutOptimizer : GraphProcessor {
     // only needs to be performed if at least one node in the previous pass is
     // expanded.
     if (graph_->node_size() > node_size_original) {
+      // Creating Const nodes holding the permutation used by added Transposes
+      // of nodes and ensure both are in the same frame.
       NodeDef* n = AddNodePermNHWCToNCHW();
       n = AddNodePermNCHWToNHWC();
+      n = n; // Avoid compilation warning
```
Comment (andyly) [inline tensorflow/core/grappler/optimizers/layout_optimizer.cc]: Does removing the assignment, as n is unused, remove the warning. i.e.

```
AddNodePermNHWCToNCHW();
AddNodePermNCHWToNHWC();
```

File: tensorflow/core/grappler/optimizers/layout_optimizer.cc [hunk 2]
```diff
@@ -2048,8 +2048,11 @@ class DataLayoutOptimizer : GraphProcessor {
     // only needs to be performed if at least one node in the previous pass is
     // expanded.
     if (graph_->node_size() > node_size_original) {
+      // Creating Const nodes holding the permutation used by added Transposes
```
Comment (andyly) [inline tensorflow/core/grappler/optimizers/layout_optimizer.cc]: Can you reword the comment to:

```
Create Const nodes holding the permutation used by added Transposes of nodes not in a frame.
```

**Comment by joyalbin** [comment]
@andyly , I have reworked on the comments, could you please review  the PR

**Review by andyly — APPROVED** [review]

**Pull request merged**